Quaver.Shared/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -527,6 +536,8 @@ public override void Update(GameTime gameTime)
             QuaverScreenChangeType type = QuaverScreenChangeType.CompleteChange)
         {
             Utils.NativeUtils.EnableWindowsKey();
+            if (IsNoFailAddedInGameplay)
+                ModManager.RemoveMod(ModIdentifier.NoFail);
```

Comment:
Removing NoFail here will make result screen try to submit the score online, you should remove it after it has checked for unranked mods.

Reply:
Fixed. Now removes NF only when retrying, quitting from pause menu and after calling SubmitOnlineScore()